agent_plugin_builder/vendor_dirs.py

```diff
@@ -186,17 +187,31 @@ def _log_container_output(container_logs: bytes, prefix: str = ""):
     logger.debug(f"{prefix} Container logs: {container_logs.decode('utf-8')}")
 
 
-def generate_requirements_file(build_dir: Path):
+def generate_requirements_file(build_dir: Path, include_hashes: bool = True):
     """
     Generate the requirements file from the lock file depending on the lock file present.
 
     :param build_dir: Path to the build directory.
+    :param include_hashes: Include hashes in the requirements file.
     """
     import subprocess
 
     logger.info("Generating requirements file")
     if (build_dir / "poetry.lock").exists():
-        command = ["poetry", "export", "-f", "requirements.txt", "-o", "requirements.txt"]
+        command = [
+            "poetry",
+            "export",
+            "--without-hashes",
+            "-f",
+            "requirements.txt",
+            "-o",
+            "requirements.txt",
+        ]
+        if include_hashes:
+            command.remove("--without-hashes")
+        else:
+            logger.warning("Hashes not included in the requirements file")
```

Comment:
Our code should be secure by default. By default, the command should exclude `--without-hashes`. We can add `--without-hashes` if the flag is set.